Lib/pickle.py

```diff
@@ -1033,31 +1033,25 @@ def _batch_appends(self, items, obj):
                 write(APPEND)
             return
 
-        it = iter(items)
         start = 0
-        while True:
-            tmp = list(islice(it, self._BATCHSIZE))
-            n = len(tmp)
-            if n > 1:
+        for batch in batched(items, self._BATCHSIZE):
+            if len(batch) != 1:
                 write(MARK)
-                for i, x in enumerate(tmp, start):
+                for i, x in enumerate(batch, start):
```

Comment:
Is it faster to calculate `start + i` in case of an error?

Reply:
It's probably faster because we don't need to materialize an `enumerate` object but I'd benchmark that one to be sure.